account_product_fiscal_classification/models/account_product_fiscal_classification.py

```diff
@@ -84,7 +84,8 @@ def write(self, vals):
         if 'purchase_tax_ids' in vals or 'sale_tax_ids' in vals:
             for fc in self:
                 pt_lst = pt_obj.browse([x.id for x in fc.product_tmpl_ids])
-                pt_lst.write({'fiscal_classification_id': fc.id})
+                if pt_lst:
```

Comment:
In other words, read the traceback of the "Expected Singleton" error to find out who overwrote the `write()` method of `product.template` and added a `self.ensure_one()`, and find a way to remove that `self.ensure_one()` call instead of adding an `if` here.